File: push_changes.py
#!/usr/bin/env python3
import subprocess
import os

os.chdir(r"C:\Users\황병하\slack-room-bot")

def run(cmd):
    result = subprocess.run(cmd, capture_output=True, text=True)
    print("CMD:", " ".join(cmd))
    if result.stdout:
        print("STDOUT:", result.stdout)
    if result.stderr:
        print("STDERR:", result.stderr)
    print("RC:", result.returncode)
    print()
    return result

print("=== GIT STATUS ===")
run(["git", "status"])

print("=== GIT REMOTE ===")
run(["git", "remote", "-v"])

print("=== GIT LOG (recent) ===")
run(["git", "log", "--oneline", "-5"])

print("=== STAGING ===")
run(["git", "add", "src/listeners/commands/help.ts"])
run(["git", "add", "src/listeners/commands/index.ts"])
run(["git", "add", "USAGE.md"])

print("=== STAGED FILES ===")
run(["git", "diff", "--cached", "--name-only"])

print("=== COMMITTING ===")
run(["git", "commit", "-m", "feat: add /사용방법 help command and usage guide"])

print("=== PUSHING ===")
run(["git", "push", "origin"])

print("=== FINAL LOG ===")
run(["git", "log", "--oneline", "-5"])

File: push_changes.mjs
import { execSync } from 'node:child_process';

const GIT = '"C:\\Users\\황병하\\AppData\\Local\\GitHubDesktop\\app-3.5.5\\resources\\app\\git\\cmd\\git.exe"';
const CWD = 'C:\\Users\\황병하\\slack-room-bot';

function run(cmd) {
  console.log(`\n>>> ${cmd}`);
  try {
    const out = execSync(cmd, { cwd: CWD, encoding: 'utf-8', stdio: ['pipe','pipe','pipe'] });
    if (out) console.log(out);
    return true;
  } catch (e) {
    if (e.stdout) console.log(e.stdout);
    if (e.stderr) console.error(e.stderr);
    console.log('EXIT:', e.status);
    return false;
  }
}

// 1. git init
console.log('===== GIT INIT =====');
run(`${GIT} init`);

// 2. remote 연결
console.log('===== ADD REMOTE =====');
run(`${GIT} remote add origin https://github.com/veronica-rona/Room-bot.git`);
run(`${GIT} remote -v`);

// 3. stage all source files
console.log('===== STAGING =====');
run(`${GIT} add src/`);
run(`${GIT} add USAGE.md`);
run(`${GIT} add package.json`);
run(`${GIT} add package-lock.json`);
run(`${GIT} add tsconfig.json`);
run(`${GIT} add .env.example`);
run(`${GIT} add .gitignore`);
run(`${GIT} add README.md`);

// 4. staged 확인
console.log('===== STAGED FILES =====');
run(`${GIT} diff --cached --name-only`);

// 5. commit
console.log('===== COMMIT =====');
run(`${GIT} commit -m "feat: add /사용방법 help command and usage guide"`);

// 6. push (main 브랜치)
console.log('===== PUSH =====');
run(`${GIT} branch -M main`);
run(`${GIT} push -u origin main`);

// 7. 결과 확인
console.log('===== RESULT =====');
run(`${GIT} log --oneline -5`);
run(`${GIT} status`);
